sktime/transformations/series/adapt.py

```diff
@@ -201,3 +201,188 @@ def get_test_params(cls, parameter_set="default"):
+
+        Parameters
+        ----------
+        X : pd.DataFrame
+            Data to fit transform to
+        y : ignored argument for interface compatibility
+            Additional data, e.g., labels for transformation
+
+        Returns
+        -------
+        self: a fitted instance of the estimator
+        """
+        if self.apply_to in ["all", "all_subset"]:
+            self._X = X
+        return self
+
+    def _transform(self, X, y=None):
+        """Transform X and return a transformed version.
+
+        private _transform containing the core logic, called from transform
+
+        Parameters
+        ----------
+        X : pd.DataFrame
+            Data to be transformed
+        y : ignored argument for interface compatibility
+            Additional data, e.g., labels for transformation
+
+        Returns
+        -------
+        Xt : pd.DataFrame
+            transformed version of X
+        """
+        apply_to = self.apply_to
+        method = self.method
+        kwargs = self.kwargs
+
+        if kwargs is None:
+            kwargs = {}
+
+        if apply_to in ["all", "all_subset"]:
+            _X = X.combine_first(self._X)
+        else:
+            _X = X
+
+        Xt = getattr(_X, method)(**kwargs)
+
+        if apply_to in "all_subset":
+            Xt = Xt.loc[X.index]
+
+        return Xt
+
+    def _update(self, X, y=None):
+        """Fit transformer to X and y.
+
+        private _fit containing the core logic, called from fit
+
+        Parameters
+        ----------
+        X : pd.DataFrame
+            Data to fit transform to
+        y : ignored argument for interface compatibility
+            Additional data, e.g., labels for transformation
+
+        Returns
+        -------
+        self: a fitted instance of the estimator
+        """
+        if self.apply_to in ["all", "all_subset"]:
+            self._X = X.combine_first(self._X)
+        return self
+
+    @classmethod
+    def get_test_params(cls, parameter_set="default"):
+        """Return testing parameter settings for the estimator.
+
+        Parameters
+        ----------
+        parameter_set : str, default="default"
+            Name of the set of test parameters to return, for use in tests. If no
+            special parameters are defined for a value, will return `"default"` set.
+
+        Returns
+        -------
+        params : dict or list of dict, default = {}
+            Parameters to create testing instances of the class
+            Each dict are parameters to construct an "interesting" test instance, i.e.,
+            `MyClass(**params)` or `MyClass(**params[i])` creates a valid test instance.
+            `create_test_instance` uses the first (or only) dictionary in `params`
+        """
+        params1 = {"method": "diff"}
+        params2 = {"method": "diff", "kwargs": {"periods": 2}, "apply_to": "all_subset"}
+
+        def add1(x):
+            return x + 1
```

Comment:
Looks like this is not used anywhere

Reply:
indeed. Wanted to use that in the `transform` case, but the pickling test fails, since functions apparenctly cannot be pickled.

Reply:
removed